crates/starlight/src/vm/value.rs

```diff
@@ -290,7 +290,7 @@ pub mod old_value {
 
 unsafe impl Trace for JsValue {
     fn trace(&mut self, visitor: &mut dyn Tracer) {
-        if self.is_object() && !self.is_empty() && !self.is_int32() {
+        if self.is_object(){
```

Comment:
you should leave `is_empty` there. Or add `is_empty` check to `is_object` since on 64 bit targets null pointer is considered as pointer too.

Reply:
`is_object` check contains !`is_empty`